### PR TITLE
Implement an example maven project

### PR DESCRIPTION
**Pull request by jin**
Add an example Java project that uses Maven for getting dependencies.

https://github.com/bazelbuild/bazel/issues/1410

I'm in the midst of writing a Skylark rule for maven_jar and maven_server,
but while that is in progress, here's the example project that I'm testing with.

### CODE REVIEW

File: java-maven/BUILD [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package(default_visibility = ["//visibility:public"])
+
+java_library(
+    name = "java-maven-lib",
+    srcs = glob(["src/main/**/myproject/*.java"]),
```
Comment (kchodorow) [inline java-maven/BUILD]: Prefer non-recursive globs (or no globs at all when there's just one file).

Comment (jin) [inline java-maven/BUILD]: done!

**Comment by kchodorow** [comment]
LGTM.

Jenkins, test this please.

**Comment by damienmg** [comment]
We do not have jenkins presubmit for this project. The reason? there is no separate configure step, we should make this project testable with only `bazel test //...` (well would be a good follow up of that change) :p

**Comment by damienmg** [comment]
LGTM too,

this project won't even run on ci.bazel.io because we only test the tutorial.

**Comment by kchodorow** [comment]
Manually built & tested, looks good.

**Pull request merged**